src/joint_storage.rs

```diff
@@ -123,3 +125,98 @@ pub fn read_joints_since_mci(db: &Connection, mci: u32) -> Result<Vec<Joint>> {
 
     Ok(joints)
 }
+
+pub fn purge_joint_and_dependencies<F>(
+    db: &mut Connection,
+    joint: &Joint,
+    err: &str,
+    f: F,
+) -> Result<()>
+where
+    F: Fn(&str, &str) + 'static,
+{
+    let unit = joint.get_unit_hash();
+    let rc_unit = Rc::new(unit.clone());
+
+    let mut tx = db.transaction()?;
+    {
+        let mut stmt =
+            tx.prepare_cached("INSERT INTO known_bad_joints (unit, json, error) VALUES (?, ?, ?)")?;
+        stmt.insert(&[unit, &serde_json::to_string(joint)?, &err])?;
+
+        let mut stmt = tx.prepare_cached("DELETE FROM unhandled_joints WHERE unit=?")?;
+        stmt.insert(&[unit])?;
+
+        let mut stmt = tx.prepare_cached("DELETE FROM dependencies WHERE unit=?")?;
+        stmt.insert(&[unit])?;
+    }
+
+    let mut queries = db::DbQueries::new();
+    storage::generate_queries_to_archive_joint(&tx, &joint, "voided", &mut queries)?;
```

Comment:
I didn't see generate_queries_to_archive_joint in the js implementation, something wrong?